circus-train-s3-mapreduce-cp/src/main/java/com/hotels/bdp/circustrain/s3mapreducecp/mapreduce/RetriableFileCopyCommand.java

```diff
@@ -159,8 +159,9 @@ private Transfer startTransfer(Mapper.Context context, S3UploadDescriptor upload
           input, uploadDescriptor.getMetadata());
       // We add 1 to the buffer size as per the com.amazonaws.RequestClientOptions doc
       request.getRequestClientOptions().setReadLimit(bufferSize + 1);
+      input.close();
```

Comment:
try with resource on line 148 maybe better?

Reply:
It seems to not work, because on line 156, `input` is reassigned another value...